fieldservice_sale_stock/models/sale_order.py

```diff
@@ -18,17 +18,18 @@ def prepare_fsm_values_for_stock_picking(self, fsm_order):
 
     def _link_pickings_to_fsm(self):
         for rec in self:
-            # TODO: We may want to split the picking to have one picking
-            # per FSM order
             fsm_order = self.env["fsm.order"].search(
                 [
                     ("sale_id", "=", rec.id),
                     ("sale_line_id", "=", False),
+                    (
+                        "stage_id",
+                        "!=",
+                        self.env.ref("fieldservice.fsm_stage_cancelled").id,
+                    ),
                 ]
             )
-            if rec.procurement_group_id:
-                rec.procurement_group_id.fsm_order_id = fsm_order.id or False
```

Comment:
Are you sure about removing these lines?

Reply:
Thank you for your comment @ivantodorovich! 

My thought process regarding the removal of these lines is to ensure that each new picking order generated by a sale order is associated with its own FSM order, rather than reusing the same FSM order for all pickings. My understanding is that the procurement_group_id ties together all stock operations related to the same sale order. By linking the procurement_group_id to an FSM order, we were enforcing a single FSM order across all related pickings.

Removing this line allowed each picking order to be independently linked to a distinct FSM order. This change ensures a clean one-to-one relationship between pickings and FSM orders.

Reply:
Will the `fsm_order_id` field be used in any other module, after this change?
If not, then maybe we need to remove it.



Additionally, I think we need to remove this related, then: https://github.com/OCA/field-service/blob/83ef6dce0e6dd72161d274c517a4a7c6d80f5d0e/fieldservice_stock/models/stock_picking.py#L9-L11

And ensure the field is populated when the picking is created

Reply:
Thank you for the feedback! I’ve applied the suggested changes, it can be tested/reviewed again. Please let me know if anything else is needed.